base/src/main/java/proguard/exception/InstructionExceptionFormatter.java

```diff
@@ -131,12 +154,20 @@ public void printException(ProguardCoreException exception, TracedVariables vari
         }
 
         // Print stack and variables
-        messageBuilder
-                .append("Variables: ")
-                .append(variables)
-                .append("\nStack: ")
-                .append(stack)
-                .append("\n");
+        if (variables != null) {
+            messageBuilder
+                    .append("Variables: ")
+                    .append(variables)
+                    .append("\n");
+        }
+
+        if (stack != null)
+        {
+            messageBuilder
+                    .append("Stack: ")
+                    .append(stack)
+                    .append("\n");
+        }
```

Comment:
Should there be a specific message if `variables` and `stack` is `null`, to show that they are empty?

Reply:
Null does not mean empty, it means we do not know them, hence I omit them.

Reply:
But previously it would show `variables: null` and now it just omits them? I think it would still be nice to have something explaining that they are omitted and why.

Reply:
Previously, it would never be `null` please note that this is a class written by us, so we know where it is called. If you still think it should be its own case, I can maybe do: `Variables: not known at this instance`

Reply:
I think if you can give an example where we can get into this case, then it would be helpful for me to understand when this would happen. But in general I think showing 'not known at this instance' seems good.

Reply:
This formatter was originally focussed on the PE, within the PE, the variables and stack are always known. Now I want to expand, so we can also use the formatter with the StackSizeComputer, this one does not know what is on the stack, only how many items are on there. Hence, from within the StackSizeComputer I can not give the variables or stack (they are not known)